src/Umbraco.Core/Persistence/Mappers/BaseMapper.cs

```diff
@@ -10,60 +8,74 @@ namespace Umbraco.Core.Persistence.Mappers
 {
     public abstract class BaseMapper : IDiscoverable
     {
-        protected BaseMapper()
-        {
-            Build();
-        }
+        // note: using a Lazy<ISqlContext> here because during installs, we are resolving the
+        // mappers way before we have a configured IUmbracoDatabaseFactory, ie way before we
+        // have an ISqlContext - this is some nasty temporal coupling which we might want to
+        // cleanup eventually.
+
+        private readonly Lazy<ISqlContext> _sqlContext;
+        private readonly object _definedLock = new object();
+        private readonly ConcurrentDictionary<Type, ConcurrentDictionary<string, string>> _maps;
 
-        internal abstract ConcurrentDictionary<string, DtoMapModel> PropertyInfoCache { get; }
+        private ISqlSyntaxProvider _sqlSyntax;
+        private bool _defined;
 
-        private void Build()
+        protected BaseMapper(Lazy<ISqlContext> sqlContext, ConcurrentDictionary<Type, ConcurrentDictionary<string, string>> maps)
```

Comment:
These are essentially breaking changes, though i don't suspect any public users are using these APIs, in either case, will need to mark this as a breaking change.